src/pal/prebuilt/idl/clrdata_i.cpp

```diff
@@ -14,15 +14,13 @@
 
 #pragma warning( disable: 4049 )  /* more than 64k source lines */
 
+#include <rpc.h>
```

Comment:
Including those header files caused templates to be exposed to non-C++ linkage.

> In file included from /home/kchoi/coreclr/src/pal/prebuilt/idl/clrdata_i.cpp:21:
In file included from /home/kchoi/coreclr/src/pal/inc/rt/rpc.h:16:
/home/kchoi/coreclr/src/pal/inc/rt/palrt.h:1250:1: error: templates must have C++ linkage
template< class T >

I wasn't sure whether there was a better fix. It looks like compiling those *_i files as C also 'works'.